### DISCUSSION TITLE
Forge, A1111 and SD.Next

### DISCUSSION THREAD

**Discussion by gohan2091** [lllyasviel/stable-diffusion-webui-forge#117] [Q&A] [answered]
I just came across Forge. SD.next and A1111 already exist, so what is different about Forge? and could the different projects not all come together as one giant team and work together on one single project? That would surely benefit the community rather than to have 3 smaller teams on 3 different systems. It just seems a shame that there is so much talent floating around and we aren't all working together in unison.

**Comment by lllyasviel** [comment, marked as answer]
**Edited Fed 10: Someone informed me that this my reply is reposted to reddit and gets controversial under an out-of-context title "FORGE is not a fork of A1111". Please note that this comment happens in the context when the original discussion is talking about SD.Next projects. Herein, the sentence `Hey we are not a fork of A1111` should be interpreted as `Hey we are not a fork of A1111 like SD.Next that will change many things and stop fetching upstream updates`. Please do not take this sentence out of its context and misunderstand it. We are not using this reply here to pretend that Forge is an independent software or some other purely original thing. In the context of this discussion, this sentence is mainly talking about the difference between Forge and the mentioned other projects. And, technically, if a repo includes all files from another repo then it should be called as a fork. All credits should go to upstream webui team for the great work.**

**Original Reply:**

Hey we are not a fork of A1111 (like SD.Next, *edited Fed 10*). Forge is a platform on top of A1111 to optimize resources and make development easier. Files from A1111 are identical and the only reason that they are inside this repo is to make installation and updating easier.

I wanted to make this repo into an extension but I failed to make the optimizations without deleting some codes from upper-stream. If I do not delete those codes, the consumed GPU memory is always 1.4GB larger than the current repo. After I removed some upper-stream codes, the 1.4GB extra GPU memory suddenly dropped. And I still have not figured out what happened. One of my guess is related to “thread abuse”, since the torch instance in upper stream in called by many different threads and there is a risk that those threads are making duplicated GPU contexts, causing extra GPU memory usage.

In theory we can make this repo like a patch installation that just download some files and also delete a few files, but that will make installation very difficult for many users. So we have to include all original files in this repo to make installation easier. But we are not going to change those files.

**Reply by deagzzz** [reply]
Your work is great, I'm really considering to switch over to it entirely but few of the extensions doesn't seem to work as someone said its build on the dev build of A1111?, I currently use the 1.6.0 version of A1111 and all the extensions do work out fine. But this forge is causing some problems so is there any way to downgrade this forge to lets say 1.6.0 version or a1111?

**Reply by gohan2091** [reply]
Some possible typos in regards to your version you currently use (which I assume is 1.7.0?). What specific extensions don't work? I am pondering whether to try this Forge out and I currently use SD.Next.

**Reply by gohan2091** [reply]
Oh I see, thanks for explaining that. What are the developers of A1111 and SD.Next reaction to Forge (if any)? While they are different, they also relate greatly. SD.Next is looking for developers and while I am by no means part of their team (nor do I wish to be to be frank), I think it would be worth considering a partnership of great minds together into one concentrated application. I think I will try out Forge sometime soon as I do prefer certain things of A1111 over SD.Next (and vice versa)

**Reply by lllyasviel** [reply]
I think it is possible to do it on 1.6.0 or even 1.5.0 but we do not have this on the to-do list rightnow

**Reply by dhmiller123** [reply]
The speed increase (on a 4600) is nearly 2x for stills. Deforum is the one extension I miss the most, but that issue was addressed by the dev elsewhere...

**Reply by ICodeDiffusion** [reply]
Forge is my new home, I get almost a 2x speed improvement on SDXL and also on SD 1.5! The performance OVERALL is incredible over any other A1111 I have used and it's been a godsend for me!
I can finally use Hiresfix for SDXL also and it's still faster than without it on the other A1111.
Incredible work!! :D
P.S I have a shipping container's worth of LoRa models and this thing handles it like a BOSS!! Like it's amazing how fast the UI performs and just everything else! like nudenet censor works flawlessly finally, I had a lot of problems with 1.7 A1111(latest version) and it would always fail part or half way into running it, and this manages to run right through very fast!
Don't get me started with Ultimate SD Upscale, it's so fast it can't even provide the preview of the process or changes, because of how fast it goes through each tile xD
